pkg/apis/sriovnetwork/v1/helper.go

```diff
@@ -87,10 +88,8 @@ func (s *SriovNetworkNodePolicySpec) Selected(iface *InterfaceExt) bool {
 	if s.NicSelector.Vendor != "" && s.NicSelector.Vendor != iface.Vendor {
 		return false
 	}
-	if s.NicSelector.DeviceID != "" {
-		if ((s.NumVfs == 0 && s.NicSelector.DeviceID != iface.DeviceID) || (s.NumVfs > 0 && s.NicSelector.DeviceID != SriovPfVfMap[iface.DeviceID])) {
-			return false
-		}
+	if s.NicSelector.DeviceID != "" && s.NicSelector.DeviceID != iface.DeviceID {
```

Comment:
Could you explain a little bit on why VF device ID works with previous code?

Reply:
In previous code, the deviceID in the ConfigMap is the same as the one defined in NodePolicy CR, and the selector will convert the PF deviceID to VF's, before comparing it to the one defined in NodePolicy CR.  It breaks the `nicSelector` semantic. Since it shall always select the PF, not the VF.